tutorials/llm/llama-3/README.rst

```diff
@@ -0,0 +1,207 @@
+that tunes a very small number of additional parameters as compared to
+full fine-tuning, thereby reducing the compute required.
+
+`NVIDIA NeMo
+Framework <https://docs.nvidia.com/nemo-framework/user-guide/latest/overview.html>`__
+provides tools to perform LoRA on Llama 3 to fit your usecase, which can
+then be deployed using `NVIDIA NIM <https://www.nvidia.com/en-us/ai/>`__
+for optimized inference on NVIDIA GPUs.
+
+.. figure:: ./img/e2e-lora-train-and-deploy.png
+  :width: 1000
+  :alt: Diagram showing the steps for customizing NVIDIA NIM with LoRA using NVIDIA NeMo. The steps include converting models to .nemo format, creating LoRA adapters with NeMo, and then using the LoRA adapter with NIM for inference on the customized model.
+  :align: center
+
+  Figure 1: Diagram representing the various steps involved in LoRA customization using NVIDIA NeMo Framework and deployment with NVIDIA NIM.
+
+
+| NIM supports seamless deployment of multiple LoRA adapters (aka “multi-LoRA”) over the same base model by dynamically loading the adapter weights based on incoming requests at runtime. This allows flexibility in serving inputs belonging to different tasks or usecases without having to deploy a bespoke model for each usecase. More information on NIM for LLMs can be found it its `documentation <https://docs.nvidia.com/nim/large-language-models latest/introduction.html>`__.
+
+Pre-requisites
+--------------
+
+In order to proceed, ensure that you have the following
+
+* System Configuration
+    * Access to at least 1 NVIDIA GPU with a cumulative memory of at least 80GB, for example: 1 x H100-80GB or 1 x A100-80GB.
+    * A docker enabled environment, with `NVIDIA Container Runtime <https://developer.nvidia.com/container-runtime>`_ installed, which will make the container GPU-aware.
+    * Other pre-requisites for NIM are mentioned `here <https://docs.nvidia.com/nim/large-language-models/latest/getting-started.html#prerequisites>`_.
+
+* Requested the necessary permission from Hugging Face and Meta to download `Meta-Llama-3-8B-Instruct <https://huggingface.co/meta-llama/Meta-Llama-3-8B-Instruct>`_. Then, you can use your Hugging Face `access token <https://huggingface.co/docs/hub/en/security-tokens>`_ to download the model, which we will then convert and customize with NeMo Framework.
+
+* `Authenticate with NVIDIA NGC <https://docs.nvidia.com/nim/large-language-models/latest/getting-started.html#ngc-authentication>`_, and download `NGC CLI Tool <https://docs.nvidia.com/nim/large-language-models/latest/getting-started.html#ngc-cli-tool>`_.
+
+
+`1. Creating a LoRA adapter with NeMo Framework <./llama3-lora-nemofw.ipynb>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This notebook showcases performing LoRA PEFT on **Llama 3 8B Instruct** using a `PubMedQA <https://pubmedqa.github.io/>`__ using NeMo Framework. PubMedQA is a Question-Answering dataset for biomedical texts.
+
```

Comment:
This notebook shows how to perform LoRA PEFT on **Llama 3 8B Instruct** using a `PubMedQA <https://pubmedqa.github.io/>`__ using NeMo Framework. PubMedQA is a Question-Answering dataset for biomedical texts. You will use the NeMo Framework which is available as a `docker
container <https://catalog.ngc.nvidia.com/orgs/nvidia/containers/nemo>`__.